code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -1095,3 +1095,84 @@
 	requirements = list(10,10,10,10,10,10,10,10,10,10)
 	logo = "gun-logo"
 	repeatable = TRUE
+
+//////////////////////////////////////////////
+//                                          //
+//       DIVERGENT CLONE                    //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/midround/from_ghosts/divergentclone
+	name = "Divergent Clone"
+	role_category = /datum/role/divergentclone
+	required_candidates = 1
+	weight = 3
+	weight_category = "Clone"
+	cost = 5
+	requirements = list(40,30,20,10,10,10,10,10,10,10)
+	high_population_requirement = 10
+	logo = "divergentclone-logo"
+	repeatable = TRUE
+	makeBody = FALSE
+	flags = MINOR_RULESET
+
+/datum/dynamic_ruleset/midround/from_ghosts/divergentclone/trim_candidates()
+	..()
+	for(var/mob/M in dead_players)
+		if(M.mind.GetRole(DIVERGENTCLONE))
+			dead_players -= M
+	for(var/mob/M in list_observers)
+		if(M.mind.GetRole(DIVERGENTCLONE))
+			list_observers -= M
+
+
+/datum/dynamic_ruleset/midround/from_ghosts/divergentclone/ready(var/forced = 0)
+	if(!config.revival_cloning)
+		return 0
+	
+	//Check that we have at least one ghost who isn't already a clone waiting to spawn
+	var/list/candies = dead_players + list_observers
+	var/list/valids[0]
+	for(var/mob/dead/observer/G in candies)
+		if(G.mind.GetRole(DIVERGENTCLONE))
+			continue
+		valids += G
+	if(valids.len == 0)
+		if(forced)
+			message_admins("Tried to force divergent clone, but no valid candidates found.")
+		return 0
+
+	var/list/clonepods = list()
+	for(var/obj/machinery/cloning/clonepod/pod in machines)
+		//Check that the pod has cloned something before
+		if(pod.cloned_records.len > 0)
+			clonepods += pod
+	if(!forced && clonepods.len == 0)
+		return 0
+	return ..()
+
+/datum/dynamic_ruleset/midround/from_ghosts/divergentclone/finish_setup(mob/new_character, index)
+	//Create a new dummy body to create a new mind for the ghost.
+	var/L = get_turf(new_character)
+	var/mob/living/carbon/human/H = new /mob/living/carbon/human(pick(latejoin))
+	H.ckey = new_character.ckey
+	H.fully_replace_character_name(null, "\improper spirit of a divergent clone")
+
+	//Give the about-to-be-a-ghost mob the provisional role and objective
+	var/datum/role/divergentclone/new_role = new /datum/role/divergentclone(H.mind, override=TRUE)
+	new_role.ForgeObjectives()
+	new_role.AnnounceObjectives()
```

Comment:
These procs should fire under /datum/role/divergentclone/New() instead of being called here.

Reply:
While I agree that might make sense, it also seems that no other role does this, so I chose not to do it here either. I'm assuming it's to allow admins to add the role to people without it instantly creating objectives or announcing the objectives to them.